lib/engine/game/g_1822_pnw/step/special_track.rb

```diff
@@ -20,8 +20,11 @@ def legal_tile_rotation?(_entity, _hex, tile)
             return true if tile.id == 'BC-0'
           end
 
-          #          def available_hex(entity, hex)
-          #          end
+          def lay_tile(action, extra_cost: 0, entity: nil, spender: nil)
+            raise GameError, 'Cannot upgrade forests' if action.hex.assigned?('forest')
```

Comment:
The forest is assigned here, but terrain in the game class?

Reply:
On the base map, hexes can contain the terrain "forest" - this is similar to hills, mountains, etc., and is put in within the map.rb file (e.g. 'upgrade=cost:10,terrain:forest', ).  Once a tile gets placed on the hex, the forest gets put on that tile, similar to how coal works in 1817 - so I patterned it after that.